pootle/apps/pootle_store/unit/timeline.py

```diff
@@ -46,172 +46,203 @@ class UnitTimelineLog(UnitLog):
+        if suggestion_accepted:
+            return SuggestionAcceptedEvent(
+                self.submission.suggestion, submission=self.submission).context
+
+        return dict(
+            value=self.submission.new_value,
+            translation=True)
 
-    entry_class = TimelineEntry
-    fields = SubmissionProxy.timeline_fields
 
-    def __init__(self, ob):
-        self.object = ob
+class UnitCreatedEvent(object):
+    def __init__(self, unit_source, **kwargs):
+        self.unit_source = unit_source
+        self.target_event = kwargs.get("target_event")
 
     @property
-    def grouped_entries(self):
-        grouped_entries = self.get_grouped_entries()
-        grouped_entries = self.add_creation_entry(grouped_entries)
-        grouped_entries.reverse()
-        return grouped_entries
+    def context(self):
+        target_changed = (
+            self.target_event is not None
+            and self.target_event.value.old_value != '')
+        ctx = dict(description=_("Unit created"))
+        if target_changed:
+            ctx['value'] = self.target_event.value.old_value
+            ctx['translation'] = True
+        else:
+            if self.unit_source.unit.istranslated():
+                ctx['value'] = self.unit_source.unit.target
+                ctx['translation'] = True
+
+        return ctx
+
+
+class UnitStateChangedEvent(SubmissionEvent):
 
     @property
-    def submissions(self):
-        submission_filter = (
-            Q(field__in=[SubmissionFields.TARGET, SubmissionFields.STATE,
-                         SubmissionFields.COMMENT, SubmissionFields.NONE]))
-        subs = (
-            Submission.objects.filter(unit=self.object)
-                              .filter(submission_filter))
-        if self.object.changed and self.object.change.commented_on:
-            subs = subs.exclude(
-                field=SubmissionFields.COMMENT,
-                creation_time=self.object.change.commented_on)
-        return subs.order_by("id")
+    def context(self):
+        params = {
+            'old_value': STATES_MAP[int(to_python(self.submission.old_value))],
+            'new_value': STATES_MAP[int(to_python(self.submission.new_value))]}
+        value = format_html(_(
+            u"%(old_value)s "
+            u"<span class='timeline-arrow'></span> "
+            u"%(new_value)s", params))
 
-    @cached_property
-    def submissions_values(self):
-        return list(self.submissions.values(*self.fields))
+        return dict(value=value,
+                    state=True)
 
+
+class CommentUpdatedEvent(SubmissionEvent):
     @property
-    def suggestion_ids(self):
-        return list(set([
-            x["suggestion_id"]
-            for x in self.submissions_values
-            if x["suggestion_id"]
-        ]))
+    def context(self):
+        if self.submission.new_value:
+            return dict(
+                value=self.submission.new_value,
+                sidetitle=_("Comment:"),
```

Comment:
I still think that this doesn't correctly tell us the action. "Added comment:" or "Set comment:" are better IMHO.